pkg/controlplane/controller/defaultservicecidr/default_servicecidr_controller.go

```diff
@@ -175,18 +190,26 @@ func (c *Controller) syncStatus(serviceCIDR *networkingapiv1.ServiceCIDR) {
 
 	// This controller will set the Ready condition to true if the Ready condition
 	// does not exist and the CIDR values match this controller CIDR values.
+	sameConfig := reflect.DeepEqual(c.cidrs, serviceCIDR.Spec.CIDRs)
 	for _, condition := range serviceCIDR.Status.Conditions {
 		if condition.Type == networkingapiv1.ServiceCIDRConditionReady {
 			if condition.Status == metav1.ConditionTrue {
+				// ServiceCIDR is Ready and config matches this apiserver
+				// nothing else is required
+				if sameConfig {
+					return
+				}
+			} else if !c.reportedNotReadyCondition {
+				klog.InfoS("default ServiceCIDR condition Ready is not True, please validate your cluster's network configuration for this ServiceCIDR", "status", condition.Status, "reason", condition.Reason, "message", condition.Message)
+				c.eventRecorder.Eventf(serviceCIDR, v1.EventTypeWarning, condition.Reason, condition.Message)
+				c.reportedNotReadyCondition = true
 				return
```

Comment:
I'm trying to understand why this is putting the early `return` on a present-and-not-ready condition inside this `!c.reportedNotReadyCondition` condition... this means the *first* time we encounter a , we log / report an event / return early... and then every subsequent sync we will *not* return early? Was that intentional?

Reply:
as an aside, I was hoping the changes in this method would be as small as possible so we could easily verify we weren't introducing new error conditions or failure modes... wrapping logging / events inside a conditional is easy to reason about... changing control flow based on new conditionals is a lot harder

Reply:
hmm this is a bug , thanks for catching it